### ISSUE TITLE
Containers selenoid/video-recorder are still not removed after finishing test session

### ISSUE THREAD

**Issue by AlexeyPaskhin** [aerokube/selenoid#597]
Enviroment:
Ubuntu 16.04.3 LTS (GNU/Linux 4.13.0-26-generic x86_64)
Docker version 18.06.0-ce, build 0ffa825
Docker API version 1.37
Selenoid "version":"1.8.1"

Logs of the Selenoid container about few not removed containers 'selenoid/video-recorder':

```
docker logs ed34ebab3ef7 2>&1 | grep -i "d3918b0fadaf\|667c9a06b3fd\|112501e44f8f\|543a28129bcf\|929569228adf\|42a5af329c34"
2018/10/31 13:45:24 [94] [STARTING_VIDEO_CONTAINER] [selenoid/video-recorder:latest-release] [42a5af329c342e50b514e630977489a25390063adfa5d6a3ee5bfd70d43c2a19]
2018/10/31 13:45:24 [94] [VIDEO_CONTAINER_STARTED] [selenoid/video-recorder:latest-release] [42a5af329c342e50b514e630977489a25390063adfa5d6a3ee5bfd70d43c2a19] [0.79s]
2018/10/31 13:46:56 [94] [STOPPING_VIDEO_CONTAINER] [42a5af329c342e50b514e630977489a25390063adfa5d6a3ee5bfd70d43c2a19]
2018/10/31 13:46:57 [94] [STOPPED_VIDEO_CONTAINER] [42a5af329c342e50b514e630977489a25390063adfa5d6a3ee5bfd70d43c2a19]
2018/10/31 15:49:13 [337] [STARTING_VIDEO_CONTAINER] [selenoid/video-recorder:latest-release] [d3918b0fadaf87092010dbe63033a343e05a5b5475a4edb10097bb3d25fe786e]
2018/10/31 15:49:13 [337] [VIDEO_CONTAINER_STARTED] [selenoid/video-recorder:latest-release] [d3918b0fadaf87092010dbe63033a343e05a5b5475a4edb10097bb3d25fe786e] [0.51s]
2018/10/31 15:49:28 [337] [STOPPING_VIDEO_CONTAINER] [d3918b0fadaf87092010dbe63033a343e05a5b5475a4edb10097bb3d25fe786e]
2018/10/31 15:49:29 [337] [STOPPED_VIDEO_CONTAINER] [d3918b0fadaf87092010dbe63033a343e05a5b5475a4edb10097bb3d25fe786e]
2018/10/31 16:03:25 [1345] [STARTING_VIDEO_CONTAINER] [selenoid/video-recorder:latest-release] [667c9a06b3fd82f824cdf725ca035c35726a84da476ca3643aee1fc0dff1232f]
2018/10/31 16:03:26 [1345] [VIDEO_CONTAINER_STARTED] [selenoid/video-recorder:latest-release] [667c9a06b3fd82f824cdf725ca035c35726a84da476ca3643aee1fc0dff1232f] [0.60s]
2018/10/31 16:03:40 [1345] [STOPPING_VIDEO_CONTAINER] [667c9a06b3fd82f824cdf725ca035c35726a84da476ca3643aee1fc0dff1232f]
2018/10/31 16:03:41 [1345] [STOPPED_VIDEO_CONTAINER] [667c9a06b3fd82f824cdf725ca035c35726a84da476ca3643aee1fc0dff1232f]
2018/10/31 16:06:03 [2021] [STARTING_VIDEO_CONTAINER] [selenoid/video-recorder:latest-release] [112501e44f8f703df5060b3a3460fc69cfa5ccfbccee209e4739224789fe8ace]
2018/10/31 16:06:03 [2021] [VIDEO_CONTAINER_STARTED] [selenoid/video-recorder:latest-release] [112501e44f8f703df5060b3a3460fc69cfa5ccfbccee209e4739224789fe8ace] [0.67s]
2018/10/31 16:06:24 [2021] [STOPPING_VIDEO_CONTAINER] [112501e44f8f703df5060b3a3460fc69cfa5ccfbccee209e4739224789fe8ace]
2018/10/31 16:06:25 [2021] [STOPPED_VIDEO_CONTAINER] [112501e44f8f703df5060b3a3460fc69cfa5ccfbccee209e4739224789fe8ace]
2018/10/31 16:17:32 [2385] [STARTING_VIDEO_CONTAINER] [selenoid/video-recorder:latest-release] [543a28129bcfdbd23fd7d2d086c527a2d8df5f75a993c91d430fdef66baf524e]
2018/10/31 16:17:33 [2385] [VIDEO_CONTAINER_STARTED] [selenoid/video-recorder:latest-release] [543a28129bcfdbd23fd7d2d086c527a2d8df5f75a993c91d430fdef66baf524e] [1.41s]
2018/10/31 16:18:08 [2385] [STOPPING_VIDEO_CONTAINER] [543a28129bcfdbd23fd7d2d086c527a2d8df5f75a993c91d430fdef66baf524e]
2018/10/31 16:18:10 [2385] [STOPPED_VIDEO_CONTAINER] [543a28129bcfdbd23fd7d2d086c527a2d8df5f75a993c91d430fdef66baf524e]
2018/10/31 16:47:47 [3525] [STARTING_VIDEO_CONTAINER] [selenoid/video-recorder:latest-release] [929569228adfa413fe77f9648d59faf37fb1219e78231a10e6569ef416ebdf8a]
2018/10/31 16:47:48 [3525] [VIDEO_CONTAINER_STARTED] [selenoid/video-recorder:latest-release] [929569228adfa413fe77f9648d59faf37fb1219e78231a10e6569ef416ebdf8a] [0.66s]
2018/10/31 16:48:07 [3525] [STOPPING_VIDEO_CONTAINER] [929569228adfa413fe77f9648d59faf37fb1219e78231a10e6569ef416ebdf8a]
2018/10/31 16:48:08 [3525] [STOPPED_VIDEO_CONTAINER] [929569228adfa413fe77f9648d59faf37fb1219e78231a10e6569ef416ebdf8a]

```

Logs of not removed 'selenoid/video-recorder' are attached

[selenoidvideo recorder logs.txt](https://github.com/aerokube/selenoid/files/2538071/selenoidvideo.recorder.logs.txt)

related to https://github.com/aerokube/selenoid/issues/552

**Comment by vania-pooh** [comment]
@AlexeyPaskhin can you confirm you have latest version of `selenoid/video-recorder:latest-release`?

**Comment by AlexeyPaskhin** [comment]
@vania-pooh   hope it is enough
![image](https://user-images.githubusercontent.com/24253248/47916458-f6bc2000-deae-11e8-9a76-1875151bda1b.png)

**Comment by AlexeyPaskhin** [comment]
> @AlexeyPaskhin can you confirm you have latest version of `selenoid/video-recorder:latest-release`?

by the way after plain upgrading of Selenoid a didn't have `selenoid/video-recorder:latest-release`, a had only `selenoid/video-recorder:latest` and could not launch my tests at all. So I pulled it explicitly by a command "docker pull selenoid/video-recorder:latest-release"

**Comment by vania-pooh** [comment]
@AlexeyPaskhin ok, very interesting. Exited containers should be automatically removed because they are launched with `--rm` flag. Could you show `docker logs` output for one of video containers?

**Comment by AlexeyPaskhin** [comment]
> @AlexeyPaskhin ok, very interesting. Exited containers should be automatically removed because they are launched with `--rm` flag. Could you show `docker logs` output for one of video containers?

```
docker logs 8da3306525fc
Waiting for display to open...
ffmpeg version 3.4 Copyright (c) 2000-2017 the FFmpeg developers
  built with gcc 6.3.0 (Alpine 6.3.0)
  configuration: --prefix=/usr --enable-avfilter --enable-gpl --enable-libx264 --disable-stripping --disable-ffplay --disable-ffprobe --disable-ffserver --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-w32threads --disable-alsa --disable-audiotoolbox --disable-cuda --disable-cuvid --disable-d3d11va --disable-dxva2 --disable-nvenc --disable-vaapi --disable-vda --disable-vdpau --disable-videotoolbox --disable-debug
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
[x11grab @ 0x55687ea6a440] Cannot get the image data event_error: response_type:0 error_code:10 sequence:9 resource_id:14680064 minor_code:4 major_code:130.
[x11grab @ 0x55687ea6a440] Stream #0: not enough frames to estimate rate; consider increasing probesize
Input #0, x11grab, from 'browser:99':
  Duration: N/A, bitrate: N/A
    Stream #0:0: Video: rawvideo (BGR[0] / 0x524742), bgr0, 1920x1080, 12 fps, 1000k tbr, 1000k tbn, 1000k tbc
Stream mapping:
  Stream #0:0 -> #0:0 (rawvideo (native) -> h264 (libx264))
Press [q] to stop, [?] for help
[libx264 @ 0x7f0013df3080] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 AVX2 LZCNT BMI2
[libx264 @ 0x7f0013df3080] profile High, level 4.0
[libx264 @ 0x7f0013df3080] 264 - core 148 - H.264/MPEG-4 AVC codec - Copyleft 2003-2016 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=12 lookahead_threads=2 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=12 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00
Output #0, mp4, to '/data/selenoidda913adf354a125a82142b680e976227.mp4':
  Metadata:
    encoder         : Lavf57.83.100
    Stream #0:0: Video: h264 (libx264) (avc1 / 0x31637661), yuv420p, 1920x1080, q=-1--1, 12 fps, 12288 tbn, 12 tbc
    Metadata:
      encoder         : Lavc57.107.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/0 buffer size: 0 vbv_delay: -1
frame=  182 fps= 12 q=-1.0 Lsize=    1128kB time=00:00:14.91 bitrate= 619.3kbits/s speed=0.964x
video:1125kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.259849%
[libx264 @ 0x7f0013df3080] frame I:4     Avg QP: 6.50  size: 46216
[libx264 @ 0x7f0013df3080] frame P:50    Avg QP:16.75  size: 17276
[libx264 @ 0x7f0013df3080] frame B:128   Avg QP:26.43  size:   801
[libx264 @ 0x7f0013df3080] consecutive B-frames:  4.4%  3.3%  6.6% 85.7%
[libx264 @ 0x7f0013df3080] mb I  I16..4: 92.3%  2.8%  4.9%
[libx264 @ 0x7f0013df3080] mb P  I16..4:  0.6%  2.5%  1.9%  P16..4:  1.6%  0.2%  0.1%  0.0%  0.0%    skip:93.1%
[libx264 @ 0x7f0013df3080] mb B  I16..4:  0.1%  0.2%  0.1%  B16..8:  1.5%  0.1%  0.0%  direct: 0.0%  skip:98.1%  L0:37.7% L1:61.6% BI: 0.7%
[libx264 @ 0x7f0013df3080] 8x8 transform intra:23.1% inter:51.2%
[libx264 @ 0x7f0013df3080] coded y,uvDC,uvAC intra: 21.7% 24.2% 22.8% inter: 0.2% 0.3% 0.1%
[libx264 @ 0x7f0013df3080] i16 v,h,dc,p: 90%  8%  1%  1%
[libx264 @ 0x7f0013df3080] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 34% 30% 23%  2%  2%  2%  2%  2%  3%
[libx264 @ 0x7f0013df3080] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 21% 27% 13%  6%  6%  7%  7%  6%  7%
[libx264 @ 0x7f0013df3080] i8c dc,h,v,p: 78% 14%  5%  3%
[libx264 @ 0x7f0013df3080] Weighted P-Frames: Y:0.0% UV:0.0%
[libx264 @ 0x7f0013df3080] ref P L0: 74.1%  3.6% 14.9%  7.4%
[libx264 @ 0x7f0013df3080] ref B L0: 82.2% 15.4%  2.4%
[libx264 @ 0x7f0013df3080] ref B L1: 92.0%  8.0%
[libx264 @ 0x7f0013df3080] kb/s:607.20
Exiting normally, received signal 15.
```

**Comment by AlexeyPaskhin** [comment]
> @AlexeyPaskhin ok, very interesting. Exited containers should be automatically removed because they are launched with `--rm` flag. Could you show `docker logs` output for one of video containers?

and in the very first message I also attached that, for another video container, and we can see these logs are quite different

**Comment by 3axap21** [comment]
Video-recorder: latest-release container stops but is not deleted.

`c:\cm>docker ps -a
CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS                        PORTS                    NAMES
d6655d098e45        selenoid/video-recorder:latest-release   "/entrypoint.sh"         20 minutes ago      Exited (255) 18 minutes ago                            quirky_bartik
73d5d99e97a0        aerokube/selenoid-ui:1.6.3               "/selenoid-ui --sele…"   22 minutes ago      Up 22 minutes                 0.0.0.0:8080->8080/tcp   selenoid-ui
fa91e49ab8cf        aerokube/selenoid:1.8.1                  "/usr/bin/selenoid -…"   22 minutes ago      Up 22 minutes                 0.0.0.0:4444->4444/tcp   selenoid
cdce96255b24        jetbrains/teamcity-server                "/run-services.sh"       3 days ago          Up 2 hours                    0.0.0.0:8111->8111/tcp   teamcity

c:\cm>docker logs d6655d098e45
Waiting for display to open...
ffmpeg version 3.4 Copyright (c) 2000-2017 the FFmpeg developers
  built with gcc 6.3.0 (Alpine 6.3.0)
  configuration: --prefix=/usr --enable-avfilter --enable-gpl --enable-libx264 --disable-stripping --disable-ffplay --disable-ffprobe --disable-ffserver --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-w32threads --disable-alsa --disable-audiotoolbox --disable-cuda --disable-cuvid --disable-d3d11va --disable-dxva2 --disable-nvenc --disable-vaapi --disable-vda --disable-vdpau --disable-videotoolbox --disable-debug
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
[x11grab @ 0x5614f43c5440] Cannot get the image data event_error: response_type:0 error_code:10 sequence:9 resource_id:14680064 minor_code:4 major_code:130.
[x11grab @ 0x5614f43c5440] Stream #0: not enough frames to estimate rate; consider increasing probesize
Input #0, x11grab, from 'browser:99':
  Duration: N/A, bitrate: N/A
    Stream #0:0: Video: rawvideo (BGR[0] / 0x524742), bgr0, 1920x1080, 24 fps, 1000k tbr, 1000k tbn, 1000k tbc
Stream mapping:
  Stream #0:0 -> #0:0 (rawvideo (native) -> h264 (libx264))
Press [q] to stop, [?] for help
[libx264 @ 0x7f0c67614080] using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 AVX2 LZCNT BMI2
[libx264 @ 0x7f0c67614080] profile High, level 4.0
[libx264 @ 0x7f0c67614080] 264 - core 148 - H.264/MPEG-4 AVC codec - Copyleft 2003-2016 - http://www.videolan.org/x264.html - options: cabac=1 ref=3 deblock=1:0:0 analyse=0x3:0x113 me=hex subme=7 psy=1 psy_rd=1.00:0.00 mixed_ref=1 me_range=16 chroma_me=1 trellis=1 8x8dct=1 cqm=0 deadzone=21,11 fast_pskip=1 chroma_qp_offset=-2 threads=7 lookahead_threads=1 sliced_threads=0 nr=0 decimate=1 interlaced=0 bluray_compat=0 constrained_intra=0 bframes=3 b_pyramid=2 b_adapt=1 b_bias=0 direct=1 weightb=1 open_gop=0 weightp=2 keyint=250 keyint_min=24 scenecut=40 intra_refresh=0 rc_lookahead=40 rc=crf mbtree=1 crf=23.0 qcomp=0.60 qpmin=0 qpmax=69 qpstep=4 ip_ratio=1.40 aq=1:1.00
Output #0, mp4, to '/data/selenoid0ed188de8e01ef9ea275c25fa1683230.mp4':
  Metadata:
    encoder         : Lavf57.83.100
    Stream #0:0: Video: h264 (libx264) (avc1 / 0x31637661), yuv420p, 1920x1080, q=-1--1, 24 fps, 12288 tbn, 24 tbc
    Metadata:
      encoder         : Lavc57.107.100 libx264
    Side data:
      cpb: bitrate max/min/avg: 0/0/0 buffer size: 0 vbv_delay: -1
frame= 2398 fps= 24 q=-1.0 Lsize=    4440kB time=00:01:39.79 bitrate= 364.5kbits/s speed=0.996x
video:4411kB audio:0kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.655281%
[libx264 @ 0x7f0c67614080] frame I:16    Avg QP:14.81  size: 81270
[libx264 @ 0x7f0c67614080] frame P:624   Avg QP:20.86  size:  4631
[libx264 @ 0x7f0c67614080] frame B:1758  Avg QP:31.41  size:   185
[libx264 @ 0x7f0c67614080] consecutive B-frames:  1.4%  2.1%  1.3% 95.2%
[libx264 @ 0x7f0c67614080] mb I  I16..4: 60.9% 25.9% 13.2%
[libx264 @ 0x7f0c67614080] mb P  I16..4:  4.1%  2.5%  0.6%  P16..4:  1.1%  0.1%  0.1%  0.0%  0.0%    skip:91.5%
[libx264 @ 0x7f0c67614080] mb B  I16..4:  0.1%  0.1%  0.0%  B16..8:  1.2%  0.0%  0.0%  direct: 0.0%  skip:98.6%  L0:42.0% L1:57.8% BI: 0.2%
[libx264 @ 0x7f0c67614080] 8x8 transform intra:33.3% inter:33.1%
[libx264 @ 0x7f0c67614080] coded y,uvDC,uvAC intra: 6.3% 5.4% 3.5% inter: 0.1% 0.1% 0.0%
[libx264 @ 0x7f0c67614080] i16 v,h,dc,p: 52% 47%  0%  0%
[libx264 @ 0x7f0c67614080] i8 v,h,dc,ddl,ddr,vr,hd,vl,hu: 43% 18% 39%  0%  0%  0%  0%  0%  0%
[libx264 @ 0x7f0c67614080] i4 v,h,dc,ddl,ddr,vr,hd,vl,hu: 29% 41% 14%  2%  2%  2%  4%  2%  4%
[libx264 @ 0x7f0c67614080] i8c dc,h,v,p: 70% 27%  3%  0%
[libx264 @ 0x7f0c67614080] Weighted P-Frames: Y:2.6% UV:2.6%
[libx264 @ 0x7f0c67614080] ref P L0: 55.8%  9.9% 28.2%  6.0%  0.0%
[libx264 @ 0x7f0c67614080] ref B L0: 64.7% 33.3%  2.0%
[libx264 @ 0x7f0c67614080] ref B L1: 96.4%  3.6%
[libx264 @ 0x7f0c67614080] kb/s:361.58
Exiting normally, received signal 15.`

**Comment by vania-pooh** [comment]
@AlexeyPaskhin this could have been fixed in Selenoid 1.8.2+ by updating Docker client to support API 1.40. Could you check whether this is still reproducing with latest release?

**Comment by vania-pooh** [comment]
```
	case <-notRunning:
```
This one could also happen when video container is in exited state.
